meshpy/mesh_creation_functions/nurbs_geometries.py

```diff
@@ -339,6 +339,71 @@ def create_nurbs_sphere_surface(radius, n_ele_u=1, n_ele_v=1):
 
     return surf
 
+def create_nurbs_hemisphere_surface(radius, n_ele_uv=1):
+    """
+    Generates a hemisphere as a NURBS surface.
+    This function constructs five segments that represent
+    the surface of a hemisphere using Non-Uniform Rational 
+    B-Splines (NURBS) based on the specified radius 
+    and the number of elements in the parametric u and 
+    v directions. To secure the connectivity between 
+    surfaces, all surfaces must have the same parametric 
+    representation on any parametric direction. Therefore, 
+    the number of elements in u- and v- directions must
+    be the same 
+
+    The function returns a list of the five sections of the
+    hemisphere. 
+
+    Args
+    ---
+    radius: double
+        radius of the hemisphere
+    n_ele_uv: int
+        number of elements in the parametric u- and v- directions
+
+    Return
+    ----
+    list: list(geomdl object)
+        a list of geomdl objects that contains the surface information
+    """
+
+    # Create the first section of the hemisphere
+    hemisphere_1 = create_nurbs_sphere_surface(radius, n_ele_u=1, n_ele_v=1)
+
+    # Create a temporary section by rotating and translating the 
+    # first section of the hemisphere
+    temp_hemisphere = operations.rotate(hemisphere_1, 90, axis=1)
+    temp_hemisphere = operations.translate(temp_hemisphere, (0, 0, - 2.0 * radius / np.sqrt(6.0) * np.sin(1.0 / 4.0 * np.pi) * 2))
+
+    # To create the hemisphere it is necessary to split the temporary 
+    # sphere section in two pieces. This split is done in u = 0.5. After 
+    # the split, the second surface is taken as it will be
+    # adjacent to the first section of the sphere
+    cut_section_sphere = operations.split_surface_u(temp_hemisphere, param=0.5)
+    hemisphere_2 = cut_section_sphere[1]
+
+    # Create the third section. Rotate and translate it accordingly
+    hemisphere_3 = operations.rotate(hemisphere_2, 90, axis=2)
+    hemisphere_3 = operations.translate(hemisphere_3, (radius * np.sin(1.0 / 4.0 * np.pi) * 2, 0, 0))
```

Comment:
Does the `operations.rotate` operation create a copy of the input patch?

Reply:
Yes, `operations.rotate` makes a copy and rotates the geomdl object.